modules/sqs/main.tf

```diff
@@ -53,17 +53,81 @@ resource "aws_sqs_queue" "metadata_queue" {
 ## ====================================================================================================
 resource "aws_sqs_queue" "staged_recovery_queue" {
   ## OPTIONAL
-  name                        = "${var.prefix}-staged-recovery-queue.fifo"
-  fifo_queue                  = true
-  content_based_deduplication = true
-  delay_seconds               = var.sqs_delay_time_seconds
-  max_message_size            = var.sqs_maximum_message_size
-  message_retention_seconds   = var.staged_recovery_queue_message_retention_time_seconds
-  tags                        = local.tags
-  policy                      = data.aws_iam_policy_document.staged_recovery_queue_policy.json
-  visibility_timeout_seconds  = 1800 # Set to double lambda max time
+  name                       = "${var.prefix}-staged-recovery-queue"
+  delay_seconds              = var.sqs_delay_time_seconds
+  max_message_size           = var.sqs_maximum_message_size
+  message_retention_seconds  = var.staged_recovery_queue_message_retention_time_seconds
+  tags                       = local.tags
+  policy                     = data.aws_iam_policy_document.staged_recovery_queue_policy.json
+  visibility_timeout_seconds = 1800 # Set to double lambda max time
+  depends_on = [
+    aws_sqs_queue.staged_recovery_deadletter_queue
+  ]
+  redrive_policy = jsonencode({
+    deadLetterTargetArn = aws_sqs_queue.staged_recovery_deadletter_queue.arn,
+    maxReceiveCount     = 3 #if number of delayed messages in the queue is greater than this number, then the messages will be sent to its DLQ.
```

Comment:
TBD with team on what number to use here. Currently I set it to 3 which means when there are >3 messages stuck in SQS, they will be sent to dead letter queue to trigger the alarm.

Reply:
According to docs, https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.htm this is actually the number of times a message can be pulled from the queue and rejects (in our case by a lambda error) before it is sent to the dead letter queue.

Reply:
I was talking about `maxReceiveCount`. See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sqs_queue#redrive_policy

Reply:
Yes. The page you linked links to the page I linked under the description for redrive_policy. From that aws doc: "The maxReceiveCount is the number of times a consumer tries receiving a message from a queue without deleting it before being moved to the dead-letter queue."

Reply:
aah good catch. fixed the description